### PR TITLE
[Test-Runner]: Updated to v3

### PR DESCRIPTION
**Pull request by meatball133**

### CODE REVIEW

File: helpers/result_to_json.cr [hunk 1]
```diff
@@ -27,8 +27,9 @@ class TestCase
   property status : String?
   property message : String?
   property output : String?
+  property task_id : Int32?
 
-  def initialize(@name, @test_code, @status, @message, @output)
+  def initialize(@name, @test_code, @status, @message, @output, @task_id = 1)
```
Comment (ErikSchierboom) [inline helpers/result_to_json.cr:32]: Shouldn't this default to `nil`? Most of the exercises won't have task IDs, as they're only used in concept exercises and I don't think Crystal has any.

File: tests/example-all-fail/expected_results.json [hunk 2]
```diff
@@ -6,7 +6,8 @@
       "name": "Bob #hey responds to stating something",
       "test_code": "Bob.hey(\"Tom-ay-to, tom-aaaah-to.\").should eq \"Whatever.\"",
       "status": "fail",
-      "message": "Expected: \"Whatever.\"      got: \"Whatevah.\""
+      "message": "Expected: \"Whatever.\"      got: \"Whatevah.\"",
+      "task_id": 1
```
Comment (ErikSchierboom) [inline tests/example-all-fail/expected_results.json:10]: There are no tasks defined for this exercise, so this should be omitted.

Comment (meatball133) [reply tests/example-all-fail/expected_results.json]: Isnt it just ensuring the result if the solution is incorrect?

Comment (ErikSchierboom) [reply tests/example-all-fail/expected_results.json]: Hmmm, I missed that nesting is already used: https://github.com/exercism/crystal-test-runner/blob/main/tests/example-all-fail/spec/mini_bob_spec.cr

Have to think about this for a bit. I'll need to do some testing.

Comment (ErikSchierboom) [reply tests/example-all-fail/expected_results.json]: Hmmm, I missed that nesting is already used: https://github.com/exercism/crystal-test-runner/blob/main/tests/example-all-fail/spec/mini_bob_spec.cr

Have to think about this for a bit. I'll need to do some testing.

Comment (meatball133) [reply tests/example-all-fail/expected_results.json]: Only 1 exercise uses that nesting atm and it is likely that exercise will shift away from it

Comment (ErikSchierboom) [reply tests/example-all-fail/expected_results.json]: Ah. For non-nested tests, does it output as `task_id` of `nil` (or omit the key)?

Comment (meatball133) [reply tests/example-all-fail/expected_results.json]: It omits the key

Comment (ErikSchierboom) [reply tests/example-all-fail/expected_results.json]: Brilliant

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**